### PR TITLE
[Docs] Sorting CODEOWNERS file.

### PR DESCRIPTION
**Pull request by rajkaramchedu**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3630)

### CODE REVIEW

**Comment by rajkaramchedu** [comment]
I am not sure if we should sort like this as it splits the commonly-owned directory paths and spreads them around. But if we still want this, I can resolve the conflict.

File: CODEOWNERS [hunk 1]
```diff
@@ -1,43 +1,61 @@
 # **Please** keep this file ordered alphabetically by directory paths!
-/aptos-move/framework/ @davidiw @movekevin @wrwg
+
+# Owners for the `/api` directory and all its subdirectories. And other API related crates
+/api/ @banool @gregnazario
+
```
Comment (JoshLind) [inline CODEOWNERS:5]: nit: can we add a comment above `/aptos-move` this so that it's consistent? 😄

**Review by JoshLind — APPROVED** [review]
LGTM! Thanks @rajkaramchedu 🚀

**Comment by JoshLind** [comment]
> I am not sure if we should sort like this as it splits the commonly-owned directory paths and spreads them around. But if we still want this, I can resolve the conflict.

FWIW, I like what you've done here. It seems more readable to me.

**Review by bowenyang007 — APPROVED** [review]

**Pull request merged**